examples/mxnet/gat/gat_batch.py

```diff
@@ -168,39 +187,55 @@ def main(args):
                 args.attn_drop,
                 args.residual)
 
-    model.initialize(ctx=ctx)
+    model.initialize()
+
+    if cuda:
+        model.collect_params().reset_ctx(ctx)
 
     # use optimizer
     trainer = gluon.Trainer(model.collect_params(), 'adam', {'learning_rate': args.lr})
 
-    # initialize graph
     dur = []
     for epoch in range(args.epochs):
         if epoch >= 3:
             t0 = time.time()
         # forward
         with mx.autograd.record():
             logits = model(features)
-            loss = mx.nd.softmax_cross_entropy(logits, labels)
-
-        #optimizer.zero_grad()
-        loss.backward()
-        trainer.step(features.shape[0])
-        loss.wait_to_read()
+            loss = mx.nd.softmax_cross_entropy(logits[mask].squeeze(), labels[mask].squeeze())
+            loss.backward()
+        trainer.step(mask.shape[0])
```

Comment:
This might be debatable @jermainewang . In gcn, batchsize in step call is set to 1. See here:
https://github.com/dmlc/dgl/blob/master/examples/mxnet/gcn/gcn.py#L180

So I am not sure whether we should rescale gradients by number of nodes here

Reply:
I 'm sure we should rescale gradients.

Reply:
Line 164-166 did take out all positions that mask != 0. The rescale should be correct. I just don't remember why we decided to use `trainer.step(batchsize=1)` in gcn. Maybe we also change GCN? @jermainewang